### PR TITLE
Initial UDP Source commit

### PR DESCRIPTION
**Pull request by AlexBlumer**
It's all working with tests. 

The only thing I absolutely want feedback on is the licensing. I'd rather keep this library as weakly licensed as possible, which is Apache without logback, but seems to be EPL/LGPL with. I have a disclaimer about customers maybe removing logback so they can reduce the license. However I'm not sure if that's how licenses, if I should just use EPL/LGPL, or remove logback and let tell them to add their own slf4j implementation.

### CODE REVIEW

File: udpSource/src/main/java/io/vantiq/extsrc/udp/MapTransformer.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
+package io.vantiq.extsrc.udp;
```
[the remark this answers is not in the document]
Comment (AlexBlumer) [reply udpSource/src/main/java/io/vantiq/extsrc/udp/MapTransformer.java]: I found something similar in your code, but it includes `SPDX: MIT`. Is that just specifying the license, and is it necessary?

[the remark this answers is not in the document]
Comment (fhcarter) [reply udpSource/src/main/java/io/vantiq/extsrc/udp/MapTransformer.java]: That is specifying the license, but no, it's not necessary (I don't think).

File: udpSource/README.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+## Licensing
+This program uses several licensed libraries. 
+okhttp3 and jackson-databind are both licensed under [Apache Version 2.0 License](http://www.apache.org/licenses/LICENSE-2.0). 
+slf4j is licensed under [terms](https://www.slf4j.org/license.html) identical to the [MIT License](https://opensource.org/licenses/MIT).
+logback is licensed under either the [Eclipse Public License v1.0](http://www.eclipse.org/legal/epl-v10.html) or [Lesser GPL v2.1](www.gnu.org/licenses/old-licenses/lgpl-2.1.html), at the licensee's discretion.
+This library will be licensed under the the Apache Version 2.0 License. HOWEVER, due to the inclusion of logback this library is covered by the licensee's choice of the more restrictive LGPL or EPL licenses. It is this author's (uninformed, unofficial, non-expert) belief that if you replace logback with a less restrictive slf4j implementation then this library will instead be covered by Apache v2.0 and any terms included in the new slf4j implementation. It is advised that you seek professional legal counsel before following the above advice.
```
[the remark this answers is not in the document]
Comment (AlexBlumer) [reply udpSource/README.md]: I might just remove the disclaimer and mention that logback is entirely replaceable, along with a how-to.

File: extjsdk/build.gradle [hunk 3]
```diff
@@ -38,6 +38,9 @@ dependencies {
     
     compile 'com.squareup.okhttp3:okhttp:3.4.1'
     compile 'com.squareup.okhttp3:okhttp-ws:3.4.1'
+    
+    compile "org.slf4j:slf4j-api:1.7.25"
+    compile 'ch.qos.logback:logback-classic:1.2.3'
```
[the remark this answers is not in the document]
Comment (AlexBlumer) [reply extjsdk/build.gradle]: For now I think I'll move it to the UDP gradle file, and then once everything else is squared away change it to something else.

**Comment by paulbutterworth** [comment]
You have to deal with the reality of the packages that are used. The best way to do this is to explicitly specify that the Vantiq copyrighted code is licensed under Apache 2.0 but that the logback library is EPL (if there is a choice between EPL and LGPL opt for EPL). As a general rule we try to stay completely away from LGPL and GPL licensed code. A number of companies are prohibited from using it.

File: udpSource/src/main/java/io/vantiq/extsrc/udp/MapTransformer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fhcarter) [reply udpSource/src/main/java/io/vantiq/extsrc/udp/MapTransformer.java]: @AlexBlumer W/r/t @paulbutterworth's  comment, you can see what I did in the LICENSE.md file in my PR #28

File: udpSource/README.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+```
+then the resulting map would be exactly the same, as the Map in lvl2 would have been overwritten by the new value.
+
+## Licensing
+The source code in this library is licensed under the the [MIT License](https://opensource.org/licenses/MIT).
+
+This library uses several licensed libraries, some of which are more restrictive than the MIT license. 
+okhttp3 and jackson-databind are both licensed under [Apache Version 2.0 License](http://www.apache.org/licenses/LICENSE-2.0). 
+slf4j is licensed under [terms](https://www.slf4j.org/license.html) identical to the [MIT License](https://opensource.org/licenses/MIT).
+logback is licensed under either the [Eclipse Public License v1.0](http://www.eclipse.org/legal/epl-v10.html) or [Lesser GPL v2.1](www.gnu.org/licenses/old-licenses/lgpl-2.1.html), at the licensee's discretion. This library is optional, and can be replaced with another slf4j implementation if you wish to reduce the license restrictions. To do so, replace the line `compile "ch.qos.logback:logback-classic:1.2.3"` with your own import, or remove that line for no logging.
```
Comment (fhcarter) [inline udpSource/README.md]: 1) ...replace the line... (need to note that you replace this in the `build.gradle` file.

Comment (AlexBlumer) [reply udpSource/README.md]: @The review comment, I moved the dependency from extjsdk to here. If I'm remembering correctly the reason that it was in exjsdk was that I wanted to see the statements in the tests so I could debug it, and then I forgot to remove it.

**Pull request merged**